pipelines/utils/execute_dbt_model/tasks.py

```diff
@@ -3,19 +3,345 @@
+                error_logs = [
+                    log_entry.get("message", "")
+                    for log_entry in result.result.logs
+                    if log_entry.get("levelname")
+                    in ("ERROR", "CRITICAL", "FATAL")
+                ]
+                if error_logs:
+                    error_details = f" Details: {'; '.join(error_logs)}"
+
+            raise FAIL(f"Failed to install dbt dependencies.{error_details}")
+
+        log("Successfully installed dbt dependencies", level="info")
+        return result
+    except Exception as e:
+        if not isinstance(e, FAIL):
+            raise FAIL(f"Error installing dbt dependencies: {str(e)}") from e
+        else:
+            raise
+
+
+@task(
+    max_retries=constants.TASK_MAX_RETRIES.value,
+    retry_delay=timedelta(seconds=constants.TASK_RETRY_DELAY.value),
+)
+def new_execute_dbt_model(
+    dbt_repository_path: str,
+    dataset_id: str,
+    table_id: Optional[str] = None,
+    dbt_alias: bool = True,
+    dbt_command: str = "run",
+    target: str = "dev",
+    flags: Optional[str] = None,
+    _vars: Optional[Union[dict, List[Dict], str]] = None,
+    disable_elementary: bool = False,
+) -> bool:
+    """
+    Execute a DBT model and process logs from the log file.
+
+    Args:
+        dbt_repository_path (str): Path to the dbt repository.
+        dataset_id (str): Dataset ID of the dbt model.
+        table_id (str, optional): Table ID of the dbt model. If None, the
+            whole dataset will be run.
+        dbt_alias (bool, optional): If True, the model will be run by
+            its alias. Defaults to True.
+        dbt_command (str, optional): The dbt command to run. Defaults to "run".
+        target (str, optional): The dbt target to use. Defaults to "dev".
+        flags (Optional[str], optional): Flags to pass to the dbt command. Defaults to None.
+        _vars (Optional[Union[dict, List[Dict], str]], optional): Variables to pass to
+            dbt. Defaults to None.
+        disable_elementary (bool, optional): Disable elementary on-run-end hooks. Defaults to False.
+
+    Raises:
+        ValueError: If dbt_command is invalid.
+        FAIL: If dbt execution fails.
+    """
+    if dbt_command not in ["run", "test", "run and test", "run/test"]:
+        raise ValueError(f"Invalid dbt_command: {dbt_command}")
+
+    if table_id:
+        if dbt_alias:
+            selected_table = f"{dataset_id}.{dataset_id}__{table_id}"
+        else:
+            selected_table = f"{dataset_id}.{table_id}"
+    else:
+        selected_table = dataset_id
+
+    if disable_elementary:
+        if _vars is None:
+            _vars = constants_execute.DISABLE_ELEMENTARY_VARS.value
+        else:
+            _vars = merge_vars(
+                constants_execute.DISABLE_ELEMENTARY_VARS.value, _vars
+            )
+
+    logs_dir = os.path.join(dbt_repository_path, "logs")
+    if not os.path.exists(logs_dir):
+        os.makedirs(logs_dir)
```

Comment:
É só uma questão de estilo, mas vc pode usar `os.makedirs(path, exist_ok = True)`  ao invés de incluir esse  `if not os.path.exists(logs_dir):`